pkg/collector/corechecks/servicediscovery/impl_linux.go

```diff
@@ -85,120 +81,53 @@ func (li *linuxImpl) DiscoverServices() (*discoveredServices, error) {
-			svc.service.Name = service.Name
-			svc.meta.Name = service.Name
-			events.heartbeat = append(events.heartbeat, *svc)
+	for _, service := range response.StoppedServices {
+		if li.ignoreCfg[service.Name] {
+			continue
 		}
-	}
 
-	// check if services previously marked as ignore are still alive.
-	for pid := range li.ignoreProcs {
-		if _, ok := serviceMap[pid]; !ok {
+		pid := service.PID
+		if li.ignoreProcs[pid] {
 			delete(li.ignoreProcs, pid)
 		}
+		events.stop = append(events.stop, service)
 	}
 
-	return &discoveredServices{
-		ignoreProcs:     li.ignoreProcs,
-		potentials:      li.potentialServices,
-		runningServices: li.aliveServices,
-		events:          events,
-	}, nil
-}
-
-// handlePotentialServices checks cached potential services we have seen in the
-// previous call of the check. If they are still alive, start events are sent
-// for these services.
-func (li *linuxImpl) handlePotentialServices(events *serviceEvents, now time.Time, serviceMap map[int]*model.Service) {
-	if len(li.potentialServices) == 0 {
-		return
-	}
-
-	// potentialServices contains processes that we scanned in the previous
-	// iteration and had open ports. We check if they are still alive in this
-	// iteration, and if so, we send a start-service telemetry event.
-	for pid, svc := range li.potentialServices {
-		if service, ok := serviceMap[pid]; ok {
-			svc.LastHeartbeat = now
-			svc.service.RSS = service.RSS
-			svc.service.CPUCores = service.CPUCores
-			svc.service.ContainerID = service.ContainerID
-			svc.service.GeneratedName = service.GeneratedName
-			svc.service.ContainerServiceName = service.ContainerServiceName
-			svc.service.ContainerServiceNameSource = service.ContainerServiceNameSource
-			svc.service.Name = service.Name
-			svc.meta.Name = service.Name
-
-			li.aliveServices[pid] = svc
-			events.start = append(events.start, *svc)
+	for _, service := range response.HeartbeatServices {
+		pid := service.PID
+		if li.ignoreProcs[pid] || li.ignoreCfg[service.Name] {
+			continue
 		}
-	}
-	clear(li.potentialServices)
-}
-
-func (li *linuxImpl) getServiceInfo(service model.Service) serviceInfo {
-	// if the process name is docker-proxy, we should talk to docker to get the process command line and env vars
-	// have to see how far this can go but not for the initial release
 
-	// for now, docker-proxy is going on the ignore list
-
-	serviceType := servicetype.Detect(service.Ports)
+		if _, ok := li.aliveServices[pid]; !ok {
+			continue
+		}
 
-	meta := ServiceMetadata{
-		Name:               service.Name,
-		Language:           service.Language,
-		Type:               string(serviceType),
-		APMInstrumentation: service.APMInstrumentation,
+		events.heartbeat = append(events.heartbeat, service)
 	}
 
-	return serviceInfo{
-		meta:          meta,
-		service:       service,
-		LastHeartbeat: li.time.Now(),
-	}
+	return &discoveredServices{
+		ignoreProcs:     li.ignoreProcs,
+		runningServices: li.aliveServices,
```

Comment:
These names are misleading right? The services being called `aliveServices` or `runningServices` are just the ones which have had the start service event emitted in this invocation.